### ISSUE TITLE
passing 11 out of 15 positive easy tests in the unit tests suite of the obsolete Java thread sanitizer

### ISSUE THREAD

**Issue by yilongli** [runtimeverification/rv-predict#146]
Threadsanitizer v2 from the dynamic testing group of Google is probably our biggest rival at the moment. The obsolete Threadsantizer v1 has an experimental version for Java which contains some interesting unit tests (~80). I believe we should include all of them in our regression test suite.

As the first step, I start with the 15 positive "easy" tests. Java Threadsantizer v1 passes 14 of them, while RV-Predict passes 11. Here are the "easy" tests: https://github.com/yilongli/java-thread-sanitizer/blob/master/tests/EasyTests.java

To run the tests:
1) clone my java-thread-sanitizer repository;
2) `ant download; ant`
3)`rv-predict -cp dist/tests.jar EasyTests`

I have added a custom `EasyTests#main` method to facilitate the running of their unit tests. Modify the method to fit your need. 

Here is the result I got:

```
yilongli@ubuntu:~/Eclipse-Workspaces/rv-workspace/java-thread-sanitizer$ rv-predict -cp dist/tests.jar EasyTests

----------------Instrumented execution to record the trace-----------------
Log directory: /tmp/rv-predict6735053937597827845
Exception in thread "Thread-58" java.lang.ArrayStoreException
    at java.lang.System.arraycopy(Native Method)
    at EasyTests$15.thread2(EasyTests.java:319)
    at ThreadRunner$2.run(ThreadRunner.java:90)

-------------------------Logging phase completed.--------------------------
Race on field EasyTests$2.sharedShort between:
    EasyTests$2.thread1(EasyTests.java:70)
    EasyTests$2.thread2(EasyTests.java:74)

Race on field EasyTests$8.sharedVar between:
    EasyTests$8.thread2(EasyTests.java:167)
    EasyTests$8.thread1(EasyTests.java:162)

Race on field EasyTests$6.sharedDouble between:
    EasyTests$6.thread2(EasyTests.java:138)
    EasyTests$6.thread1(EasyTests.java:134)

Race on field EasyTests$7.sharedObject between:
    EasyTests$7.thread2(EasyTests.java:152)
    EasyTests$7.thread1(EasyTests.java:148)

Race on field EasyTests$1.sharedVar between:
    EasyTests$1.thread2(EasyTests.java:58)
    EasyTests$1.thread1(EasyTests.java:54)

Race on field EasyTests$10.sharedVar between:
    EasyTests$10.thread1(EasyTests.java:194)
    EasyTests$10.thread2(EasyTests.java:200)

Race on field EasyTests$3.sharedChar between:
    EasyTests$3.thread2(EasyTests.java:90)
    EasyTests$3.thread1(EasyTests.java:86)

Race on field EasyTests$9.sharedVar between:
    EasyTests$9.thread2(EasyTests.java:182)
    EasyTests$9.thread1(EasyTests.java:178)

Race on field EasyTests$4.sharedLong between:
    EasyTests$4.thread2(EasyTests.java:106)
    EasyTests$4.thread1(EasyTests.java:102)

Race on field EasyTests$11$RecursiveObject.value between:
    EasyTests$11.thread2(EasyTests.java:230)
    EasyTests$11.thread1(EasyTests.java:226)

Race on field EasyTests$5.sharedFloat between:
    EasyTests$5.thread2(EasyTests.java:122)
    EasyTests$5.thread1(EasyTests.java:118)
```

Basically, RV-Predict failed the four tests starting with `systemArrayCopy`.

@traiansf would you like to work on them? I could also work on them as a way to study the system. But it could take a while.

BTW, I only convert the java-thread-sanitizer part of the Threadsanitizer v1 from googlecode svn to github, so you won't be able to run java-thread-sanitizer using that repository. Even if you check out the complete Threadsanitizer v1, java-thread-sanitizer would only work with certain version of jdk 1.6 (probably because of its dependency on asm-3.2). Openjdk 1.6.30 on the rv machine seems to work. Openjdk 1.6.45 on my laptop doesn't.
